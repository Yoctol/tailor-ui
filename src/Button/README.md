# Button Docs

```js
import { Button } from "@yoctol/ui";
```

```jsx
<Button>Text</Button>
```

Try it with `knobs`:

<!-- STORY -->

## API

| Props      | type                                                                              | Default | Description          |
| ---------- | --------------------------------------------------------------------------------- | ------- | -------------------- |
| `block`    | Boolean                                                                           | `false` |                      |
| `circle`   | Boolean                                                                           | `false` |                      |
| `active`   | Boolean                                                                           | `false` |                      |
| `ghost`    | Boolean                                                                           | `false` |                      |
| `light`    | Boolean                                                                           | `false` |                      |
| `disabled` | Boolean                                                                           | `false` |                      |
| `size`     | `'sm'`&#124;`'m'`&#124;`'lg'`                                                     | `'m'`   |                      |
| Space      | [View on docs](https://github.com/jxnblk/styled-system/blob/master/docs/table.md) |         | From `styled-stsyem` |
