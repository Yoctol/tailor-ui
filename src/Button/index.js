import PropTypes from 'prop-types';
import React, { PureComponent, createRef } from 'react';
import styled, { css, keyframes } from 'styled-components';
import { ifProp, switchProp } from 'styled-tools';
import { space, themeGet } from 'styled-system';

import controlTransition from '../utils/transition';

import Ripple from './Ripple';

const spin = keyframes`
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
`;

const StyledButton = styled.button`
  display: inline-flex;
  position: relative;
  align-items: center;
  justify-content: center;
  overflow: hidden;
  border: ${themeGet('borders.default')};
  border-radius: ${themeGet('radii.1')};
  border-color: ${themeGet('colors.primary')};
  background-color: ${themeGet('colors.primaryDark')};
  color: ${themeGet('colors.light')};
  line-height: ${themeGet('lineHeight')};
  text-decoration: none;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;

  &:focus {
    border-color: ${themeGet('colors.primaryDark')};
    outline: 0;
  }

  &:hover {
    background-color: ${themeGet('colors.light')};
    color: ${themeGet('colors.primaryDark')};
  }

  &:disabled,
  &[disabled] {
    opacity: 0.5;
    cursor: default;
    pointer-events: none;
  }

  ${switchProp('size', {
    sm: css`
      padding: ${themeGet('space.paddingYSm')} ${themeGet('space.paddingXSm')};
      font-size: ${themeGet('fontSizes.sm')};
    `,
    m: css`
      padding: ${themeGet('space.paddingY')} ${themeGet('space.paddingX')};
      font-size: ${themeGet('fontSizes.default')};
    `,
    lg: css`
      padding: ${themeGet('space.paddingYLg')} ${themeGet('space.paddingXLg')};
      font-size: ${themeGet('fontSizes.lg')};
    `,
  })}

  ${ifProp(
    'block',
    css`
      width: 100%;
    `
  )}

  ${ifProp(
    'danger',
    css`
      border-color: ${themeGet('colors.error')};
      background-color: ${themeGet('colors.error')};
      color: ${themeGet('colors.light')};

      &:focus {
        border-color: ${themeGet('colors.error')};
      }

      &:hover {
        background-color: ${themeGet('colors.error')};
        color: ${themeGet('colors.light')};
      }
    `
  )};

  ${ifProp(
    'light',
    css`
      border-color: ${themeGet('colors.gray.8')};
      background-color: ${themeGet('colors.light')};
      color: ${themeGet('colors.gray.2')};

      &:focus {
        border-color: ${themeGet('colors.primaryDark')};
      }

      &:hover {
        background-color: ${themeGet('colors.gray.8')};
      }

      ${ifProp(
        'active',
        css`
          border-color: ${themeGet('colors.secondaryDark')};
          background-color: ${themeGet('colors.secondaryDark')};
          color: ${themeGet('colors.light')};

          &:hover {
            background-color: ${themeGet('colors.secondary')};
            color: ${themeGet('colors.light')};
          }

          &:focus {
            border-color: ${themeGet('colors.secondaryDark')};
          }
        `
      )};
    `
  )};

  ${ifProp(
    'circle',
    css`
    border-radius: 999px;

    ${ifProp(
      { size: 'sm' },
      css`
        padding: ${themeGet('space.paddingYSm')}
          calc(${themeGet('space.paddingXSm')} * 2);
      `
    )}
    ${ifProp(
      { size: 'm' },
      css`
        padding: ${themeGet('space.paddingY')}
          calc(${themeGet('space.paddingX')} * 2);
      `
    )}
    ${ifProp(
      { size: 'lg' },
      css`
        padding: ${themeGet('space.paddingYLg')}
          calc(${themeGet('space.paddingXLg')} * 2);
      `
    )}
  `
  )}

  ${ifProp(
    'ghost',
    css`
      border-color: ${themeGet('colors.light')};
      background-color: transparent;
    `
  )}

  ${ifProp(
    'loading',
    css`
      color: transparent;
      pointer-events: none;

      &::after {
        content: '';
        display: block;
        position: absolute;
        top: calc(50% - (1em / 2));
        left: calc(50% - (1em / 2));
        width: 1em;
        height: 1em;
        border: 2px solid ${themeGet('colors.gray.6')};
        border-radius: 50%;
        border-top-color: transparent;
        border-right-color: transparent;
        animation: ${spin} 0.7s linear infinite;
      }
    `
  )}

  ${controlTransition()};

  ${space};
`;

class Button extends PureComponent {
  constructor(props) {
    super(props);
    this.ripple = createRef();
    this.button = createRef();
  }

  handleClick = e => this.ripple.current.startRipple(e, this.button);

  render() {
    const { children, ...props } = this.props;
    return (
      <StyledButton
        innerRef={this.button}
        onMouseUp={this.handleClick}
        onTouchend={this.handleClick}
        {...props}
      >
        {children}
        <Ripple ref={this.ripple} />
      </StyledButton>
    );
  }
}

Button.displayName = 'Button';

Button.propTypes = {
  /**
   * Description of prop "active".
   */
  active: PropTypes.bool,
  /**
   * Description of prop "block".
   */
  block: PropTypes.bool,
  /**
   * Description of prop "circle".
   */
  circle: PropTypes.bool,
  /**
   * Description of prop "ghost".
   */
  danger: PropTypes.bool,
  /**
   * Description of prop "light".
   */
  ghost: PropTypes.bool,
  /**
   * Description of prop "danger".
   */
  light: PropTypes.bool,
  /**
   * Description of prop "loading".
   */
  loading: PropTypes.bool,
  /**
   * Description of prop "size".
   */
  size: PropTypes.oneOf(['sm', 'm', 'lg']),
  /**
   * Description of prop "Space".
   */
  ...space.propTypes,
};

Button.defaultProps = {
  size: 'm',
  block: false,
  circle: false,
  danger: false,
  ghost: false,
  light: false,
  active: false,
  loading: false,
};

export default Button;
