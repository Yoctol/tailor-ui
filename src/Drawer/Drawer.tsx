import React, { FunctionComponent, ReactNode, useEffect } from 'react';
import { Transition, animated, config } from 'react-spring';

import Portal from 'utils/Portal';
import styled, { css } from 'utils/styled-components';
import tag from 'utils/CleanTag';
import useKeydown, { ESC_KEY_CODE } from 'utils/useKeydown';

import Backdrop from '../Backdrop';
import { FooterWrapper, ModalContent, ModalHeader } from '../Modal';

type Placement = 'top' | 'right' | 'bottom' | 'left';

interface IDrawerWrapperProps {
  breadth: string;
  placement: Placement;
}

const DrawerWrapper = styled<IDrawerWrapperProps, 'div'>(tag.div)`
  display: flex;
  position: absolute;
  z-index: 10000;
  flex-direction: column;
  background-color: #fff;

  ${p =>
    ['top', 'bottom'].includes(p.placement)
      ? css`
          top: ${p.placement === 'top' ? 0 : `calc(100vh - ${p.breadth})`};
          height: ${p.breadth};
          width: 100vw;
          box-shadow: 0 ${p.placement === 'bottom' && '-'}2px 8px
            rgba(0, 0, 0, 0.15);
        `
      : css`
          left: ${p.placement === 'left' ? 0 : `calc(100vw - ${p.breadth})`};
          width: ${p.breadth};
          height: 100vh;
          box-shadow: ${p.placement === 'right' && '-'}2px 0 8px
            rgba(0, 0, 0, 0.15);
        `}
`;

const formatBreadth = (breadth: number | string) =>
  typeof breadth === 'number' ? `${breadth}px` : breadth;

const getTransformAxis = (placement: Placement) => {
  if (['top', 'bottom'].includes(placement)) {
    return 'translateY';
  }

  return 'translateX';
};

const getTransformBreadth = ({
  placement,
  breadth,
}: {
  placement: Placement;
  breadth: string;
}) => {
  const prefix = ['left', 'top'].includes(placement) ? '-' : '';
  return `${prefix}${breadth}`;
};

const getWrapperBreadth = ({
  placement,
  width,
  height,
}: {
  placement: Placement;
  width: number | string;
  height: number | string;
}) => {
  if (['top', 'bottom'].includes(placement)) {
    return formatBreadth(height);
  }

  return formatBreadth(width);
};

export interface IDrawerProps {
  visible: boolean;
  onClose: () => void;
  title?: string;
  footer?: ReactNode;
  placement?: Placement;
  closable?: boolean;
  maskClosable?: boolean;
  width?: string | number;
  height?: string | number;
}

const Drawer: FunctionComponent<IDrawerProps> = ({
  visible,
  title,
  footer,
  placement = 'right',
  width = 400,
  height = 300,
  closable = true,
  maskClosable = true,
  onClose,
  children,
  ...otherProps
}) => {
  const breadth = getWrapperBreadth({ placement, width, height });
  const transformAxis = getTransformAxis(placement);
  const transformBreadth = getTransformBreadth({ placement, breadth });

  useKeydown({
    listening: visible,
    keyCode: ESC_KEY_CODE,
    onKeydown: onClose,
  });

  useEffect(
    () => {
      document.body.style.overflow = visible ? 'hidden' : '';

      return () => {
        document.body.style.overflow = '';
      };
    },
    [visible]
  );

  return (
    <Portal appendFor="drawer">
      <Transition
        native
        items={visible}
        from={{
          offset: transformBreadth,
        }}
        enter={{
          offset: '0px',
        }}
        leave={{
          offset: transformBreadth,
        }}
        config={{
          ...config.default,
          precision: 0.1,
        }}
      >
        {_visible =>
          _visible &&
          ((props: { offset: any }) => (
            <>
              <Backdrop
                visible={visible}
                onClick={() => {
                  if (maskClosable) {
                    onClose();
                  }
                }}
              />
              <animated.div
                style={{
                  transform: props.offset.interpolate(
                    (offset: any) => `${transformAxis}(${offset})`
                  ),
                }}
              >
                <DrawerWrapper placement={placement} breadth={breadth}>
                  {title && (
                    <ModalHeader
                      title={title}
                      onCancel={onClose}
                      closable={closable}
                    />
                  )}
                  <ModalContent {...otherProps}>{children}</ModalContent>
                  {footer && <FooterWrapper>{footer}</FooterWrapper>}
                </DrawerWrapper>
              </animated.div>
            </>
          ))
        }
      </Transition>
    </Portal>
  );
};

export default Drawer;
