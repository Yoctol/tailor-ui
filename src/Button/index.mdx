---
name: Button
menu: General
route: /general/button
---

import { PropsTable, Playground } from 'docz';
import { MdCheck, MdDelete, MdClose, MdSearch } from 'react-icons/md';
import { Toggle } from 'react-powerplug';

import Box from '../Grid/Box';
import Icon from '../Icon';

import Button from './';

# Button

To trigger an operation.

## When To Use

A button means an operation (or a series of operations). Clicking a button will trigger corresponding business logic.

## Examples

```js
import { Button } from '@yoctol/ui';
```

### Basic

<Playground>
  <Button>Basic Button</Button>
  <Button text ml="2">Text Button</Button>
</Playground>

### Type

<Playground>
  <Button type="primary">Primary</Button>
  <Button type="secondary" ml="2">
    Secondary
  </Button>
  <Button type="info" ml="2">
    Info
  </Button>
  <Button type="success" ml="2">
    Success
  </Button>
  <Button type="warning" ml="2">
    Warning
  </Button>
  <Button type="danger" ml="2">
    Danger
  </Button>
</Playground>

### Rounded

<Playground>
  <Button type="primary" rounded>Primary</Button>
  <Button type="secondary" ml="2" rounded>
    Secondary
  </Button>
  <Button type="info" ml="2" rounded>
    Info
  </Button>
  <Button type="success" ml="2" rounded>
    Success
  </Button>
  <Button type="warning" ml="2" rounded>
    Warning
  </Button>
  <Button type="danger" ml="2" rounded>
    Danger
  </Button>
</Playground>

### Outlined

<Playground>
  <Button type="primary" outlined>Primary</Button>
  <Button type="secondary" ml="2" outlined>
    Secondary
  </Button>
  <Button type="info" ml="2" outlined>
    Info
  </Button>
  <Button type="success" ml="2" outlined>
    Success
  </Button>
  <Button type="warning" ml="2" outlined>
    Warning
  </Button>
  <Button type="danger" ml="2" outlined>
    Danger
  </Button>
</Playground>

### Rounded & Outlined

<Playground>
  <Button type="primary" rounded outlined>Primary</Button>
  <Button type="secondary" ml="2" rounded outlined>
    Secondary
  </Button>
  <Button type="info" ml="2" rounded outlined>
    Info
  </Button>
  <Button type="success" ml="2" rounded outlined>
    Success
  </Button>
  <Button type="warning" ml="2" rounded outlined>
    Warning
  </Button>
  <Button type="danger" ml="2" rounded outlined>
    Danger
  </Button>
</Playground>

### Icon

```js
import { MdCheck, MdDelete, MdClose, MdSearch } from 'react-icons/md';
```

<Playground>
  <Button type="primary" icon={MdCheck}>Confirm</Button>
  <Button type="danger" icon={MdDelete} ml="2">Delete</Button>
  <Button type="info" icon={MdSearch} ml="2">Search</Button>
  <br />
  <br />
  <Button type="primary" rounded icon={MdSearch} ml="2" />
  <Button type="danger" rounded icon={MdDelete} ml="2" />
  <Button type="info" rounded icon={MdSearch} ml="2" />
  <br />
  <br />
  <Button>
    <Icon type={MdCheck} fill="success" size="20" mr="2" />
    Success
  </Button>
  <Button ml="2">
    <Icon type={MdClose} fill="danger" size="20" mr="2" />
    Danger
  </Button>
</Playground>


### Disabled

<Playground>
  <Button type="primary" disabled>Primary</Button>
  <Button type="secondary" ml="2" disabled>
    Secondary
  </Button>
  <Button type="info" ml="2" disabled>
    Info
  </Button>
  <Button type="success" ml="2" disabled>
    Success
  </Button>
  <Button type="warning" ml="2" disabled>
    Warning
  </Button>
  <Button type="danger" ml="2" disabled>
    Danger
  </Button>
</Playground>

### Loading

<Playground>
  <Button type="primary" loading>Primary</Button>
  <Button type="secondary" ml="2" loading>
    Secondary
  </Button>
  <Button type="info" ml="2" loading>
    Info
  </Button>
  <Button type="success" ml="2" loading>
    Success
  </Button>
  <Button type="warning" ml="2" loading>
    Warning
  </Button>
  <Button type="danger" ml="2" loading>
    Danger
  </Button>
  <br />
  <br />
  <Toggle>
    {({ on, toggle }) => (
      <Button
        loading={on}
        onClick={() => {
          toggle()
          setTimeout(toggle, 3000)
        }}
      >
        Click me!
      </Button>
    )}
  </Toggle>
</Playground>

### Sizes

<Playground>
  <Button size="sm">
    Button sm
  </Button>

  <Button ml={2}>
    Button md
  </Button>

  <Button ml={2} size="lg">
    Button lg
  </Button>
</Playground>

### Block

<Playground>
  <Button block>
    Block Button
  </Button>
</Playground>

## API

<PropsTable of={Button} />
