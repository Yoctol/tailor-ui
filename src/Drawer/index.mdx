---
name: Drawer
menu: Components
route: /components/drawer
---

import { PropsTable, Playground } from 'docz';

import Button from '../Button';
import Radio from '../Radio';

import Drawer from './';

# Drawer

Panel slides from screen edge.

## When To Use

A Drawer is a panel that is typically overlaid on top of a page and slides in from the side. It contains a set of information or actions. Since that user can interact with the Drawer without leaving the current page, tasks can be achieved more efficient within the same context.

- Use a Form to create or edit a set of information.
- Processing subtasks. When subtasks are too heavy for Popover and we still want to keep the subtasks in the context of the main task, Drawer - comes very handy.

When a same Form is needed in multiple places.

## Examples

```js
import { Drawer } from 'tailor-ui';
```

### Basic

<Playground>
  {() => {
    const [visible, setVisible] = React.useState(false);

    return (
      <>
        <Button onClick={() => setVisible(!visible)}>Open</Button>
        <Drawer
          title="Title"
          visible={visible}
          onClose={() => setVisible(false)}
        >
          Content...
        </Drawer>
      </>
    )

}}

</Playground>

### Custom Placement

<Playground>
  {() => {
    const [visible, setVisible] = React.useState(false);
    const [placement, setPlacement] = React.useState('left');

    return (
      <>
        <Radio.Group value={placement} onChange={setPlacement}>
          <Radio value="top">Top</Radio>
          <Radio value="right">Right (default)</Radio>
          <Radio value="bottom">Bottom</Radio>
          <Radio value="left">Left</Radio>
        </Radio.Group>

        <Button mt="2" onClick={() => setVisible(!visible)}>Open</Button>

        <Drawer
          title="Title"
          visible={visible}
          placement={placement}
          onClose={() => setVisible(false)}
        >
          Content...
        </Drawer>
      </>
    )

}}

</Playground>

### Add Custom Footer

<Playground>
  {() => {
    const [visible, setVisible] = React.useState(false);

    return (
      <>
        <Button onClick={() => setVisible(!visible)}>Open</Button>
        <Drawer
          title="Title"
          visible={visible}
          onClose={() => setVisible(false)}
          footer={
            <Button type="primary" onClick={() => setVisible(false)}>Close Drawer</Button>
          }
        >
          Content...
        </Drawer>
      </>
    )

}}

</Playground>

### hidden close button and prevent mask clicked close

<Playground>
  {() => {
    const [visible, setVisible] = React.useState(false);

    return (
      <>
        <Button onClick={() => setVisible(!visible)}>Open</Button>
        <Drawer
          title="Title"
          visible={visible}
          onClose={() => setVisible(false)}
          closable={false}
          maskClosable={false}
          footer={
            <Button type="primary" onClick={() => setVisible(false)}>Close Drawer</Button>
          }
        >
          Content...
        </Drawer>
      </>
    )

}}

</Playground>

## API

<PropsTable of={Drawer} />
