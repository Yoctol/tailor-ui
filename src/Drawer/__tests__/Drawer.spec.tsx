import React from 'react';

import { render } from 'test/test-utils';

import { Button } from '../../Button';
import { Drawer } from '../Drawer';

describe('Drawer', () => {
  it('should render correctly', () => {
    const { baseElement } = render(
      <Drawer title="Title" visible onClose={() => {}}>
        Content
      </Drawer>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render correctly with props closable', () => {
    const { baseElement } = render(
      <Drawer title="Title" visible closable={false} onClose={() => {}}>
        Content
      </Drawer>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render custom footer correctly', () => {
    const { baseElement } = render(
      <Drawer
        title="Title"
        visible
        closable={false}
        onClose={() => {}}
        footer={<Button variant="primary">Close Drawer</Button>}
      >
        Content
      </Drawer>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
