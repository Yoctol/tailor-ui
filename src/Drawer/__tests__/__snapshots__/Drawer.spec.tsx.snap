// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should render correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  cursor: default;
  margin-right: 0rem;
}

.c8 svg {
  fill: #67758d;
  vertical-align: middle;
  pointer-events: none;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c8.c8.c8 svg {
  width: 24px;
  height: 24px;
}

.c4 {
  font-size: 1.25rem;
  margin: 0;
  color: #283040;
  font-weight: bold;
  line-height: 1.5;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  border: 1px solid;
  border-radius: 0.5rem;
  border-color: transparent;
  line-height: 1.5;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 999px;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
  width: 2.25rem;
  height: 2.25rem;
  background-color: #eef1f8;
  color: #3990cd;
  font-weight: bold;
}

.c6:focus {
  outline: 0;
}

.c6:disabled,
.c6[disabled] {
  opacity: 0.85;
  background-color: #c4cfdd;
  color: #67758d;
  cursor: default;
  pointer-events: none;
}

.c6:disabled .c7 svg,
.c6[disabled] .c7 svg {
  fill: #67758d;
}

.c6::before {
  content: '';
  display: inline-block;
  position: relative;
  left: 0;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 0;
  height: 0;
  margin-right: 0;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c6 .c7 svg {
  fill: #3990cd;
}

.c6:hover {
  background-color: #f9fafc;
}

.c6:active {
  background-color: #e3e8f4;
}

.c0 {
  position: fixed;
  z-index: -1;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0,0,0,0.65);
}

.c5 {
  position: absolute;
  top: 24px;
  right: 32px;
  background: transparent;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.c5:hover {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  left: calc(100vw - 400px);
  width: 400px;
  height: 100vh;
  box-shadow: -2px 0 8px rgba(0,0,0,0.15);
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  right: 10px;
  bottom: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<body
  style="overflow: hidden; padding-right: 0px;"
>
  <div />
  <div
    portal-container=""
  >
    <div
      style="position: fixed; top: 0px; left: 0px; z-index: 20;"
    >
      <div
        class="c0"
        style="opacity: 1;"
      />
    </div>
    <div
      style="position: fixed; top: 0px; left: 0px; z-index: 20;"
    >
      <div
        style="transform: translateX(0px);"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <h3
                class="c4"
              >
                Title
              </h3>
            </div>
            <div
              class="c5"
            >
              <button
                class="c6"
              >
                <i
                  class="c7 c8"
                  cursor="default"
                  size="24"
                  style="pointer-events: none;"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </i>
              </button>
            </div>
          </div>
          <div
            class="c9"
          >
            Content
          </div>
        </div>
      </div>
    </div>
    <div
      style="position: fixed; top: 0px; left: 0px; z-index: 30;"
    >
      <div
        class="c10"
      />
    </div>
  </div>
</body>
`;

exports[`Drawer should render correctly with props closable 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    portal-container=""
  >
    .c0 {
  position: fixed;
  z-index: -1;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0,0,0,0.65);
}

<div
      style="position: fixed; top: 0px; left: 0px; z-index: 20;"
    >
      <div
        class="c0"
        style="opacity: 1;"
      />
    </div>
    .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
}

.c3 {
  font-size: 1.25rem;
  margin: 0;
  color: #283040;
  font-weight: bold;
  line-height: 1.5;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  left: calc(100vw - 400px);
  width: 400px;
  height: 100vh;
  box-shadow: -2px 0 8px rgba(0,0,0,0.15);
}

<div
      style="position: fixed; top: 0px; left: 0px; z-index: 20;"
    >
      <div
        style="transform: translateX(0px);"
      >
        <div
          class="c0"
        >
          <div
            class="c1"
          >
            <div
              class="c2"
            >
              <h3
                class="c3"
              >
                Title
              </h3>
            </div>
          </div>
          <div
            class="c4"
          >
            Content
          </div>
        </div>
      </div>
    </div>
    .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  right: 10px;
  bottom: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
      style="position: fixed; top: 0px; left: 0px; z-index: 30;"
    >
      <div
        class="c0"
      />
    </div>
  </div>
  <div />
</body>
`;

exports[`Drawer should render custom footer correctly 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    portal-container=""
  >
    .c0 {
  position: fixed;
  z-index: -1;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0,0,0,0.65);
}

<div
      style="position: fixed; top: 0px; left: 0px; z-index: 20;"
    >
      <div
        class="c0"
        style="opacity: 1;"
      />
    </div>
    .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: #eef1f8;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.c3 {
  font-size: 1.25rem;
  margin: 0;
  color: #283040;
  font-weight: bold;
  line-height: 1.5;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  border: 1px solid;
  border-radius: 0.5rem;
  border-color: transparent;
  line-height: 1.5;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
  height: 2.25rem;
  padding: 0 0.75rem;
  font-size: 1rem;
  background-color: #328ef5;
  color: #ffffff;
  font-weight: bold;
}

.c6:focus {
  outline: 0;
}

.c6:disabled,
.c6[disabled] {
  opacity: 0.85;
  background-color: #c4cfdd;
  color: #67758d;
  cursor: default;
  pointer-events: none;
}

.c6:disabled .sc-10g5g0q-0 svg,
.c6[disabled] .sc-10g5g0q-0 svg {
  fill: #67758d;
}

.c6::before {
  content: '';
  display: inline-block;
  position: relative;
  left: 0;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 0;
  height: 0;
  margin-right: 0;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c6 .sc-10g5g0q-0 svg {
  fill: #ffffff;
}

.c6:hover {
  background-color: #4196f6;
}

.c6:active {
  background-color: #2386f4;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  left: calc(100vw - 400px);
  width: 400px;
  height: 100vh;
  box-shadow: -2px 0 8px rgba(0,0,0,0.15);
}

<div
      style="position: fixed; top: 0px; left: 0px; z-index: 20;"
    >
      <div
        style="transform: translateX(0px);"
      >
        <div
          class="c0"
        >
          <div
            class="c1"
          >
            <div
              class="c2"
            >
              <h3
                class="c3"
              >
                Title
              </h3>
            </div>
          </div>
          <div
            class="c4"
          >
            Content
          </div>
          <div
            class="c5"
          >
            <button
              class="c6"
            >
              Close Drawer
            </button>
          </div>
        </div>
      </div>
    </div>
    .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  right: 10px;
  bottom: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
      style="position: fixed; top: 0px; left: 0px; z-index: 30;"
    >
      <div
        class="c0"
      />
    </div>
  </div>
  <div />
</body>
`;
